### ISSUE TITLE
Align api with @rescript/react

### ISSUE THREAD

**Issue by alexfedoseev** [bloodyowl/rescript-react-update#8]
@bloodyowl I was showing ReScript React code to engineers unfamiliar with ReScript and they raised 2 conserns they found confusing. These concerns make sense to me since these parts were inherited from ReasonReact and don't align well with React.js and official ReScript bindings.

1. React docs use `dispatch` name with `useReducer` hook. But this package uses `send` name.
2. Order of arguments in reducer function of this package is flipped relative to React. 

What do you think about updating API to match ReScript bindings to React?

**Comment by bloodyowl** [comment]
Yeah, the initial goal was to match the record API to facilitate the transition. I guess we could draft a breaking release.

**Comment by sudo97** [comment]
Same question, can I join?

**Comment by bloodyowl** [comment]
Sure @sudo97! You're more than welcome.

I guess the best course of action for now would be to only touch `ReactUpdate` and  keep `ReactUpdateLegacy` as is:
- to add a `dispatch` alias in the `self` type, that'd hold the same value as the `send` field
- to make `useReducer` have the following signature:

```rescript
let useReducer: (('state, 'action) => update<'action, 'state>, 'state) => ('state, 'action => unit)
let useReducerWithMapState: (('state, 'action) => update<'action, 'state>, unit => 'state) => ('state, 'action => unit)
```

**Comment by sudo97** [comment]
@bloodyowl I will also update some examples to be in Rescript, and not in reason

**Comment by sudo97** [comment]
@bloodyowl what do you think about I get rid of `bsb` and install fresher `rescript`?

**Comment by bloodyowl** [comment]
I guess that's ok yeah, maybe just in a separate PR

**Comment by bloodyowl** [comment]
Landed in 4.0.0, thanks @sudo97!
